noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -119,38 +133,47 @@ fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_sha2
     }
     shared_secret_bytes_with_separator[32] = GENERATOR_INDEX__SYMMETRIC_KEY;
 
-    sha256::digest(shared_secret_bytes_with_separator)
+    let mut result: [[u8; 32]; N] = std::mem::zeroed();
+
+    result[0] = sha256::digest(shared_secret_bytes_with_separator);
```

Comment:
This would be wrong if `N` were not 1

Reply:
Refactored the ugly genericism from the sha256 function

Reply:
Removed sha256 fns